src/main/java/de/tum/in/www1/artemis/service/icl/GitPublickeyAuthenticatorService.java

```diff
@@ -56,6 +64,26 @@ public boolean authenticate(String username, PublicKey publicKey, ServerSession
                 log.error("Failed to convert stored public key string to PublicKey object", e);
             }
         }
+        else if (localCIBuildJobQueueService.isPresent() && localCIBuildJobQueueService.get().getBuildAgentInformation().stream().anyMatch(agent -> {
+            if (agent.publicSshKey() == null) {
+                return false;
+            }
+
+            AuthorizedKeyEntry agentKeyEntry = AuthorizedKeyEntry.parseAuthorizedKeyEntry(agent.publicSshKey());
+            PublicKey agentPublicKey;
+            try {
+                agentPublicKey = agentKeyEntry.resolvePublicKey(null, null, null);
+            }
+            catch (IOException | GeneralSecurityException e) {
+                return false;
+            }
+
+            return agentPublicKey.equals(publicKey);
+        })) {
```

Comment:
A multi-line lambda inside of an if expression is IMO pretty bad code style. I think it would make sense to extract this lambda into a private method.